reel-picker/src/main/java/com/innowisegroup/reelpicker/datetime/LocalDateTime.kt

```diff
@@ -35,38 +30,56 @@ class LocalDateTime private constructor(private val date: LocalDate, private val
         ) {
-            if (!isDateWithinMinMaxValue(
-                    initialLocalTime.toLocalDate(),
-                    minLocalDateTime.toLocalDate(),
-                    maxLocalDateTime.toLocalDate()
-                )
-            ) throw IllegalArgumentException(
-                "Initial date value must be within max and min bounds"
+            checkValidDate(
+                initialLocalDateTime.toLocalDate(),
+                minLocalDateTime.toLocalDate(),
+                maxLocalDateTime.toLocalDate()
             )
-            if (initialLocalTime == minLocalDateTime) {
-                if (!isTimeWithinMinMaxValue(
-                        initialLocalTime.toLocalTime(),
-                        minLocalDateTime.toLocalTime(),
-                        LocalTime.of(MAX_HOUR, MAX_MINUTE)
-                    )
-                ) throw IllegalArgumentException(
-                    "Initial time value must be within max and min bounds"
-                )
-            }
-            if (initialLocalTime == maxLocalDateTime) {
-                if (!isTimeWithinMinMaxValue(
-                        initialLocalTime.toLocalTime(),
-                        LocalTime.of(MIN_HOUR, MIN_MINUTE),
-                        maxLocalDateTime.toLocalTime()
-                    )
-                ) throw IllegalArgumentException(
-                    "Initial time value must be within max and min bounds"
-                )
-            }
+            checkValidTime(
+                initialLocalDateTime.toLocalTime(),
+                minLocalDateTime.toLocalTime(),
+                maxLocalDateTime.toLocalTime()
+            )
+        }
+
+        private fun checkValidDate(
+            initialLocalDate: LocalDate,
+            minLocalDate: LocalDate,
+            maxLocalDate: LocalDate
+        ) {
+            require(initialLocalDate.year >= minLocalDate.year)
+            { "minLocalDate must not be more than initialLocalDate" }
+            require((initialLocalDate.month >= minLocalDate.month) || (initialLocalDate.year != minLocalDate.year))
+            { "minLocalDate must not be more than initialLocalDate" }
+            require((initialLocalDate.day >= minLocalDate.day) || (initialLocalDate.month != minLocalDate.month) || (initialLocalDate.year != minLocalDate.year))
+            { "minLocalDate must not be more than initialLocalDate" }
+
+            require(initialLocalDate.year <= maxLocalDate.year)
+            { "maxLocalDate must not be less than initialLocalDate" }
+            require((initialLocalDate.month <= maxLocalDate.month) || (initialLocalDate.year != maxLocalDate.year))
+            { "maxLocalDate must not be less than initialLocalDate" }
+            require(((initialLocalDate.day <= maxLocalDate.day) || (initialLocalDate.month != maxLocalDate.month) || (initialLocalDate.year != minLocalDate.year)))
+            { "maxLocalDate must not be less than initialLocalDate" }
```

Comment:
require(
                initialLocalDate.year >= minLocalDate.year || initialLocalDate.year <= maxLocalDate.year ||
                    initialLocalDate.month >= minLocalDate.month || initialLocalDate.month <= maxLocalDate.month ||
                    initialLocalDate.day >= minLocalDate.day || initialLocalDate.day <= maxLocalDate.day
            ) { "initialLocalDate does not fit min..max range of minLocalDate and maxLocalDate" }